lua/sp.c

```diff
@@ -3124,13 +3207,14 @@ static int dbstmt_bind(Lua L)
     return dbstmt_bind_int(L, lua_touserdata(L, 1));
 }
 
-static inline void setup_first_sqlite_step(SP sp, dbstmt_t *dbstmt)
+static inline void setup_first_sqlite_step(SP sp, dbstmt_t *dbstmt, int profile)
```

Comment:
Add comments indicating that non-zero profile parameter indicates that external stepping is not being used and we must denote the start of iterating over the result set.